utils/loggers/__init__.py

```diff
@@ -238,10 +240,20 @@ def on_val_end(self, nt, tp, fp, p, r, f1, ap, ap50, ap_class, confusion_matrix)
                 self.clearml.log_debug_samples(files, title='Validation')
             if self.mlflow:
                 [self.mlflow.log_artifacts(f, "validation") for f in files if f.exists()]
-
+            
         if self.comet_logger:
             self.comet_logger.on_val_end(nt, tp, fp, p, r, f1, ap, ap50, ap_class, confusion_matrix)
 
+    def on_val_end_compute_metrics(self, names, ap50, ap, f1, p, r):
+        # log metrics for each class for every epoch
+        if self.mlflow:
+            for i in range(0, len(names)):
+                self.mlflow.log_metric(f"{names[i]}_mAP_0.5", ap50[i])
+                self.mlflow.log_metric(f"{names[i]}_mAP_0.5-0.95", ap[i])
+                self.mlflow.log_metric(f"{names[i]}_f1-score", f1[i])
+                self.mlflow.log_metric(f"{names[i]}_precision", p[i])
+                self.mlflow.log_metric(f"{names[i]}_recall", r[i])
+
     def on_fit_epoch_end(self, vals, epoch, best_fitness, fi):
```

Comment:
might need to handle `on_val_end` similar to how comet is handling it. Gonna implement that too.